modyn/selector/internal/selector_strategies/abstract_downsample_strategy.py

```diff
@@ -23,11 +24,51 @@ class AbstractDownsampleStrategy(AbstractPresampleStrategy):
     def __init__(self, config: dict, modyn_config: dict, pipeline_id: int, maximum_keys_in_memory: int):
         super().__init__(config, modyn_config, pipeline_id, maximum_keys_in_memory)
 
-        if "downsampled_batch_size" not in self._config:
-            raise ValueError("To use downsampling strategies, you have to specify the downsampled_batch_size")
-        self.downsampled_batch_size = self._config["downsampled_batch_size"]
+        if "sample_then_batch" not in config:
+            raise ValueError(
+                "Please specify if you want to sample and then batch or vice versa. "
+                "Use the sample_then_batch parameter"
+            )
+        if config["sample_then_batch"]:
+            self.downsampling_mode = DownsamplingMode.SAMPLE_THEN_BATCH
+        else:
+            self.downsampling_mode = DownsamplingMode.BATCH_THEN_SAMPLE
 
-        if not isinstance(self.downsampled_batch_size, int):
-            raise ValueError("The downsampled batch size must be an integer")
+        if self.downsampling_mode == DownsamplingMode.BATCH_THEN_SAMPLE and "downsampling_period" in config:
+            raise ValueError("downsampling_period can be used only in sample-then-batch.")
+
+        self.downsampling_period = config.get("downsampling_period", 1)
+
+        if "downsampling_ratio" not in config:
+            raise ValueError("Please specify downsampling_ratio to use downsampling methods")
+
+        self.downsampling_ratio = config["downsampling_ratio"]
+
+        if not (0 < self.downsampling_ratio < 100) or not isinstance(self.downsampling_ratio, int):
+            raise ValueError("The downsampling ratio must be an integer in (0,100)")
 
         self._requires_remote_computation = True
+
+    def get_training_status_bar_scale(self) -> int:
```

Comment:
One last question about this: Why can't we just hide the bar in BtS, but need to show a bar of size 100% instead?

Reply:
The bar in BTS is never started because you never receive a massage with is_training = False (so its not hidden)

Basically, the problem is knowing how to size the two bars when they are different sizes. In BTS this is not a problem, in STB it is.